babashka: 0.0.78 -> 0.0.89

bennyandresen opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Version bump

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@AndersonTorres

bhougland18 commented:
Thanks @bandresen! I just saw this yesterday.  This version has some big enhancements.

FRidh commented:
@GrahamcOfBorg build babashka

FRidh commented:
@GrahamcOfBorg build babashka

bhougland18 commented:
Any possibility to bump it to 0.0.8 instead?

bandresen commented:
I sill don't know why @GrahamcOfBorg doesn't build this (or clj-kondo) but it compiles fine on my system.

bandresen commented:
Maybe @GrahamcOfBorg doesn't try to build because graalvm8 has not been build successfully hydra in quite awhile. (Got this from a comment in #84354 by @jlesquembre

bandresen commented:
for the next reviewer:

according to @hlolli in #83080 the package can't be built because it gets OOM-killed in grahamofborg's build.
I measured the peak memory resource consumption at 25GB when I built it myself.

It builds fine on my system and could be included, as the situation is the same for 0.0.78 or 0.0.71 before it.

bandresen commented:
Updated to latest available babashka version at this point in time (0.0.88-2)

bennyandresen committed:
babashka: utf8 + nativeBuildinputs fix

utf8: see borkdude/babashka/issues/359
nativeBuildinputs: this PR's review

bandresen commented:
Also pushed a commit for the utf-8 error that was found and fixed by @bfortz in #85393

hlolli commented:
4wiw, I'm still working hard on getting graalvm8/11 working on v20. These statically compiled .a libraries needed to run the jvmci8 is giving me a very hard time. But I hope to have a PR open soon.

hlolli commented:
Sorry not jvmci8, but substratevm expects jdk home with jvmci8 (in graal8's case) with .a libraries, and those have to come from jdk8 hotspot. With oracle using their own build farm to produce these files (ref. https://github.com/graalvm/openjdk8-jvmci-builder/issues/11 ) it makes this bit difficult path to get right.

Mic92 commented on pkgs/development/interpreters/clojure/babashka.nix:
```diff
@@ -1,24 +1,25 @@
 
-  buildInputs = [ graalvm8 ];
+  LC_ALL = "en_US.UTF-8";
+  buildInputs = [ graalvm8 glibcLocales ];
```
`glibcLocales` is used at build time. Hence it should be put into `nativeBuildInputs`.

Mic92 replied:
Same applies to `graalvm8`:

```suggestion
  nativeBuildInputs = [ graalvm8 glibcLocales ];
```

bandresen replied:
Corrected, thanks!

bennyandresen committed:
babashka: 0.0.78 -> 0.0.89

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/137

bennyandresen committed:
babashka: increase java heap space max to 4500

3g has very long build times and also doesn't build 100% of the time
it's also the new default since:
https://github.com/borkdude/babashka/commit/7fc33dcb21c067ea8212e07a55e3ac08c284409f

Mic92 commented:
Result of `nixpkgs-review pr 84350` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- babashka
</details>
